src/PackageJsonSynchronizer.php

```diff
@@ -21,15 +21,37 @@
 class PackageJsonSynchronizer
 {
     private $rootDir;
+    private $vendorDirname;
 
-    public function __construct(?string $rootDir)
+    public function __construct(string $rootDir, string $vendorDirname)
     {
         $this->rootDir = $rootDir;
+        $this->vendorDirname = $vendorDirname;
     }
 
-    public function shouldSynchronize(): bool
+    public function shouldSynchronize(array $packagesNames): bool
     {
-        return $this->rootDir && file_exists($this->rootDir.'/package.json');
+        if (!$this->rootDir || !file_exists($this->rootDir.'/package.json')) {
+            return false;
+        }
+
+        foreach ($packagesNames as $packageName) {
+            if (!$this->resolveAssetsDir($packageName)) {
+                continue;
+            }
+
+            if (!file_exists($this->rootDir.'/'.$this->vendorDirname.'/'.$packageName.'/composer.json')) {
+                continue;
+            }
+
+            $package = @json_decode(file_get_contents($this->rootDir.'/'.$this->vendorDirname.'/'.$packageName.'/composer.json'), true);
+
+            if (\is_array($package) && \in_array('symfony-ux', $package['keywords'] ?? [], true)) {
```

Comment:
Is it possible to create a "symfony-ux" package type instead of using a keyword ?

Reply:
It was discussed (I don't remember in which issue, @nicolas-grekas do you remember) and chose to do this way.

Reply:
Right now, we use the tag, but the plan was to turn this into a package type and to require this package type to look for controllers. Something in the TODO

Reply:
A package type is initially meant to control the composer installer being used, and each package can have only one such type.
A package shipping a symfony-ux component is likely to also ship other things in the code (most of the time a symfony bundle). And the management of the symfony-ux logic does not rely on a composer installer. So to me, there is no benefit to using the package type rather than a keyword. It only adds more restrictions about how you can use them in packages (and as said before, it does not change the discoverability on packagist.org as both type and keywords can be used as search filters).

Side note: Flex used to enforce a `symfony-bundle` type to register bundles in the past, and this was found to be too restrictive. Why redoing the same pattern ?

Reply:
Oh, I thought you were the one advocating for using package type.
Fine to me then of course! We would still require the keyword to look for controllers, isn't it?

Reply:
I've always been the advocating *against* using the package type.

And yes, I think requiring the keyword makes sense. It will make sure such packages can be discovered through the keyword.